backend/src/db/migrations/20200207080200-fulltext_index_for_tags.js

```diff
@@ -0,0 +1,51 @@
+import { getDriver } from '../../db/neo4j'
+
+export const description =
+  'This migration adds a fulltext index for the tags in order to search for Hasthags.'
+
+export async function up(next) {
+  const driver = getDriver()
+  const session = driver.session()
+  const transaction = session.beginTransaction()
+
+  try {
+    await transaction.run(`
+      CALL db.index.fulltext.createNodeIndex("tag_fulltext_search",["Tag"],["id"])
+    `)
+    await transaction.commit()
+    next()
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.log(error)
+    await transaction.rollback()
+    // eslint-disable-next-line no-console
+    console.log('rolled back')
+    throw new Error(error)
+  } finally {
+    session.close()
+  }
+}
+
+export async function down(next) {
+  const driver = getDriver()
+  const session = driver.session()
+  const transaction = session.beginTransaction()
+
+  try {
+    // Implement your migration here.
+    await transaction.run(`
+      CALL db.index.fulltext.drop("tag_fulltext_search")
+    `)
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-20T11:23:32Z" title="Friday, March 20th 2020, 12:23:32 pm +01:00">Mar 20, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,51 @@
+import { getDriver } from '../../db/neo4j'
+
+export const description =
+  'This migration adds a fulltext index for the tags in order to search for Hasthags.'
+
+export async function up(next) {
+  const driver = getDriver()
+  const session = driver.session()
+  const transaction = session.beginTransaction()
+
+  try {
+    await transaction.run(`
+      CALL db.index.fulltext.createNodeIndex("tag_fulltext_search",["Tag"],["id"])
+    `)
+    await transaction.commit()
+    next()
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.log(error)
+    await transaction.rollback()
+    // eslint-disable-next-line no-console
+    console.log('rolled back')
+    throw new Error(error)
+  } finally {
+    session.close()
+  }
+}
+
+export async function down(next) {
+  const driver = getDriver()
+  const session = driver.session()
+  const transaction = session.beginTransaction()
+
+  try {
+    // Implement your migration here.
+    await transaction.run(`
+      CALL db.index.fulltext.drop("tag_fulltext_search")
+    `)
```


I really like that you are adding the `down` function as well. I think I haven't added it on every migration, and some are easier to add then others, but just thought I'd point out that this is a great habit!